main.nf

```diff
@@ -0,0 +1,932 @@
+    file '.command.log' into bwa_stdout
+
+
+    script:
+    rg="\'@RG\\tID:${params.run_id}\\tSM:${params.run_id}\\tPL:illumina\'"
+    if(params.singleEnd){
+    """
+         bwa mem \\
+        -R $rg \\
+        -t ${task.cpus} \\
+        -k 2 \\
+        $params.gfasta \\
+        $reads \\
+            > ${name}_bwa.sam
+
+    # Print version number to standard out
+    echo "BWA Version:"\$(bwa 2>&1)
+    """
+    } else {
+    """
+        bwa mem \\
+        -R $rg \\
+        -t ${task.cpus} \\
+        -k 2 \\
+        $params.gfasta \\
+        $reads\\
+            > ${name}_bwa.sam
+
+    # Print version number to standard out
+    echo "BWA Version:"\$(bwa 2>&1)
+    """
+
+}
+}
+
+/*
+*  STEP 3 - Convert to BAM, sort BAM  
+*/ 
+
+process sortSam {
+    tag "${name}"
+    publishDir "${params.outdir}/BWAmem", mode: 'copy',
+        saveAs: {filename -> params.saveAlignedIntermediates ? "$filename" : null }
+    
+    input:
+    set val(name), file(raw_sam) from raw_aln_sam
+    
+
+    output: 
+    set val(name), file("${raw_sam}.sorted.bam") into samples_sorted_bam
+
+    script:
+    def avail_mem = task.memory == null ? '' : "-m ${Math.floor(task.memory.toBytes() / task.cpus)}"
+    """
+    samtools sort \\
+        $raw_sam \\
+        -@ ${task.cpus} $avail_mem \\
+        -o ${raw_sam}.sorted.bam
+    """
+}
+
+/*
+*  STEP 4 - Mark PCR duplicates in sorted BAM file
+*/ 
+
+process markDuplicates {
+    tag "${name}"
+    publishDir "${params.outdir}/markDup/metrics", mode: 'copy',
+        saveAs: { filename -> filename.indexOf(".dup_metrics") > 0 ? filename : null }
+
+    input:
+    set val(name), file(sorted_bam) from samples_sorted_bam
+
+    output:
+    set val(name), file("${name}_markdup.bam") into samples_markdup_bam
+    file("${name}.dup_metrics") into markdup_results
+    file '.command.log' into markDuplicates_stdout
+
+    script:
+    """
+        picard MarkDuplicates \\
+        INPUT=$sorted_bam \\
+        OUTPUT=${name}_markdup.bam \\
+        METRICS_FILE=${name}.dup_metrics \\
+        TMP_DIR=tmp \\
+        VALIDATION_STRINGENCY=SILENT \\
+        REMOVE_DUPLICATES=false \\
+        ASSUME_SORTED=false \\
+        MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP=50000 \\
+        MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=8000 \\
+        SORTING_COLLECTION_SIZE_RATIO=0.25 \\
+        READ_NAME_REGEX=\"[a-zA-Z0-9]+:[0-9]:([0-9]+):([0-9]+):([0-9]+).*\" \\
+        OPTICAL_DUPLICATE_PIXEL_DISTANCE=100 \\
+        VERBOSITY=INFO \\
+        QUIET=false \\
+        COMPRESSION_LEVEL=5 \\
+        CREATE_INDEX=false \\
+        MAX_RECORDS_IN_RAM=500000 \\
+        CREATE_MD5_FILE=false \\
+        GA4GH_CLIENT_SECRETS=''
+
+    # Print version number to standard out
+    echo "Picard version "\$(picard  MarkDuplicates --version 2>&1)
+    """
+}
+
+
+/*
+ * Step 5 - Recalibrate BAM file with known variants and BaseRecalibrator
+ * 
+*/ 
+process recal_bam_files {
+    tag "${name}"
```

Comment:
Missing a `publishDir` statement here.

Reply:
I added one in 9ccc5da64c9ac6f16f9c95c30e64a9db9e35015c